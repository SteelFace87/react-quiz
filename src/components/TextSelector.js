
import React from 'react';
import PropTypes from 'prop-types';

export default function TextInputForm({ text, handleTextChange, handleSubmit }){

  

  return (
    <form onSubmit={handleSubmit}>
      <input handleSubmit={handleSubmit} onChange={handleTextChange} value={text}></input>
      <button>submit</button> 
    </form>
  );
}

TextInputForm.propTypes = {
  text: PropTypes.string,
  handleTextChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}
;
