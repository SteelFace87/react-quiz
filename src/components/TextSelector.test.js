import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector', ()=>{
  it('renders stuff', ()=>{
    const wrapper =  shallow(<TextSelector text='test' handleTextChange={()=>'hi'} handleSubmit={()=>'bye'}/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
