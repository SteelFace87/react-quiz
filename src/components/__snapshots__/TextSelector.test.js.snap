// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector renders stuff 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    handleSubmit={[Function]}
    onChange={[Function]}
    value="test"
  />
  <button>
    submit
  </button>
</form>
`;
